component/http/middleware/middleware.go

```diff
@@ -122,6 +124,21 @@ func NewRecovery() Func {
 	}
 }
 
+// NewAppNameVersion returns to all responses a header which contain the app name and version.
+func NewAppNameVersion(name, version string) Func {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if version != "" {
+				w.Header().Set(appVersionHeader, version)
+			}
+			if name != "" {
+				w.Header().Set(appNameHeader, name)
+			}
+			next.ServeHTTP(w, r)
+		})
+	}
```

Comment:
I'm wondering whether we need the empty string checks within the `handlerFunc` based on the assumption that are both mandatory in our context (since we already have a check in the OptionFunc helper).

If we want to keep the checks, I think it would be more clear (and also more efficient) to move this check before the return func statement and return an error if something is wrong.

_These checks could even be completely skipped since the param check also happens by the caller before enabling the middleware, but we can keep it as a best practice since the func does not trust the caller. Also, the other middlewares don't seem to do any error checking_ 

```suggestion
func NewAppNameVersion(name, version string) Func, error {
        if version != "" {
		return nil, errors.New(...invalid version...)
	}
	if name != "" {
		return nil, errors.New(...invalid name...)
	}
	return func(next http.Handler) http.Handler {
		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
			w.Header().Set(appVersionHeader, version)
			w.Header().Set(appNameHeader, name)
			next.ServeHTTP(w, r)
		})
	}
```

Reply:
The proposed solutoin would breack the middleware signature i am afraid if we return an error.
It would make the usage a two step approach. 
First create the middleware then check for error and use it in the route vs just add it to the route.
If you check out all other middleware they all are not doing and defensive programming.

Reply:
> The proposed solutoin would breack the middleware signature i am afraid if we return an error.

It actually won't break the signature since the middleware interface is maintained in the 
`return func(next http.Handler) http.Handler {..}` statement.
The caller will need to check for the error before enabling/including the middleware in the pipeline

> If you check out all other middleware they all are not doing and defensive programming.

This is related to my second point in the above comment. If I'm not missing something, they don't do any error checking. For example, `Compression` does not validate the deflate level,  the `RequestObserver` does not validate the method and the path etc. I think that's ok btw, as long as we are consistent on how we approach this with all of our middlewares.

Reply:
Could it make sense to discuss this separately? Open a ticket to discuss this separately?
If we do error handling we need to change every middleware we have built ,which will be a breaking change since they are exported.

Reply:
Yes, as discussed we will be working on this in a separate issue to manage all middleware errors